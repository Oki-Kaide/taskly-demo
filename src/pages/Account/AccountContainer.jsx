import React from 'react';
import PropTypes from 'prop-types';
import Account from './Account';
import ProtonSDK from '../../utils/proton';

class AccountContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      completedTasks: [
        'Filed your tax return',
        'Organized all your contacts',
        'Contact DMV about updating tags',
        'Sent a request to your building manager',
      ],
    };
  }

  /* istanbul ignore next */
  openConfirmModal = async () => {
    const { actor, permission, isPageHidden, setErrorState } = this.props;
    const actions = [
      {
        account: 'xtokens',
        name: 'transfer',
        authorization: [
          {
            actor: actor,
            permission: permission,
          },
        ],
        data: {
          from: actor,
          to: ProtonSDK.requestAccount,
          quantity: '5.000000 FOOBAR',
          memo: 'Taskly',
        },
      },
    ];
    const { processed, error } = await ProtonSDK.sendTransaction(actions);
    if (error) {
      setErrorState(error);
      return;
    }
    if (!processed || !processed.id) {
      return;
    }

    if (isPageHidden()) {
      window.onfocus = this.loadTasksPage;
    } else {
      this.loadTasksPage();
    }
  };

  loadTasksPage = () => {
    const { history } = this.props;
    history.push('/tasks');
    window.onfocus = null;
  };

  loadTasksPage = () => {
    const { history } = this.props;
    history.push('/tasks');
    window.onfocus = null;
  }

  render() {
    const { completedTasks } = this.state;
    const { accountData, logout, error } = this.props;

    return (
      <Account
        openConfirmModal={this.openConfirmModal}
        completedTasks={completedTasks}
        accountData={accountData}
        logout={logout}
        error={error}
      />
    );
  }
}

AccountContainer.propTypes = {
  accountData: PropTypes.shape(),
  permission: PropTypes.string,
  actor: PropTypes.string,
  logout: PropTypes.func.isRequired,
  isPageHidden: PropTypes.func.isRequired,
};

AccountContainer.defaultProps = {
  accountData: {},
  permission: '',
  actor: '',
};

export default AccountContainer;
