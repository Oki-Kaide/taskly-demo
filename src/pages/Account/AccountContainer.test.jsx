import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AccountContainer from './AccountContainer';

const createTestProps = props => ({
  logout: jest.fn(),
  isPageHidden: jest.fn(),
  actor: 'test',
  permission: 'active',
  history: {
    push: jest.fn(),
  },
  ...props,
});

describe('AccountContainer component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<AccountContainer {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<AccountContainer {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
