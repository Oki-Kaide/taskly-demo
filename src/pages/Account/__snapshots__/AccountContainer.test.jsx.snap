// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountContainer component matches the rendered snapshot 1`] = `
<Account
  accountData={Object {}}
  completedTasks={
    Array [
      "Filed your tax return",
      "Organized all your contacts",
      "Contact DMV about updating tags",
      "Sent a request to your building manager",
    ]
  }
  logout={[MockFunction]}
  openConfirmModal={[Function]}
/>
`;
